service/rpc/servicer.go

```diff
@@ -419,6 +419,320 @@ func (s *Servicer) BlockTransaction(ctx context.Context, request *types.BlockTra
+			err := errors.New("failed to parse 'address' as common.Address")
+			return nil, LogErrValidation(err), err
+		}
+
+		callOpts, err := buildCallOpsFromCallRequest(ctx, callRequest)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		registry, err := registry.New(s.cc)
+		account, err := registry.GetAccountsContract(ctx, nil)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		voteSigner, err := account.GetVoteSigner(callOpts, address)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		result := map[string]interface{}{
+			"voteSigner": voteSigner,
+		}
+
+		return &types.CallResponse{
+			Result:     result,
+			Idempotent: true,
+		}, nil, nil
+
+	case canReceiveVotes:
+		group, ok := callRequest.Parameters["group"].(common.Address)
+		if !ok {
+			err := errors.New("failed to parse 'group' as common.Address")
+			return nil, LogErrValidation(err), err
+		}
+
+		votes, ok := callRequest.Parameters["votes"].(*big.Int)
+		if !ok {
+			err := errors.New("failed to parse 'votes' as *big.Int")
+			return nil, LogErrValidation(err), err
+		}
+
+		callOpts, err := buildCallOpsFromCallRequest(ctx, callRequest)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		registry, err := registry.New(s.cc)
+		election, err := registry.GetElectionContract(ctx, nil)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+
+		canReceiveVotes, err := election.CanReceiveVotes(callOpts, group, votes)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		result := map[string]interface{}{
+			"canReceiveVotes": canReceiveVotes,
+		}
+
+		return &types.CallResponse{
+			Result:     result,
+			Idempotent: true,
+		}, nil, nil
+
+	case getEpochNumber:
+		callOpts, err := buildCallOpsFromCallRequest(ctx, callRequest)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		registry, err := registry.New(s.cc)
+		election, err := registry.GetElectionContract(ctx, nil)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		epochNumber, err := election.GetEpochNumber(callOpts)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		result := map[string]interface{}{
+			"epochNumber": epochNumber,
+		}
+
+		return &types.CallResponse{
+			Result:     result,
+			Idempotent: true,
+		}, nil, nil
+
+	case getEpochSize:
+		callOpts, err := buildCallOpsFromCallRequest(ctx, callRequest)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		registry, err := registry.New(s.cc)
+		election, err := registry.GetElectionContract(ctx, nil)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		size, err := election.GetEpochSize(callOpts)
+		if err != nil {
+			return nil, LogErrValidation(err), err
+		}
+
+		result := map[string]interface{}{
+			"size": size,
+		}
+
+		return &types.CallResponse{
+			Result:     result,
+			Idempotent: true,
+		}, nil, nil
+
+	case getEpochNumberOfBlocks:
```

Comment:
does this, `getEpochNumber` and `getEpochSize` all need to be separate? Seems like they are closely coupled, and you could return them in one call? https://github.com/celo-org/celo-monorepo/blob/819a92da2aa69eb7eb868968b802291ccfbdee98/packages/protocol/contracts/common/UsingPrecompiles.sol#L55

I don't know how these functions are being used though

Reply:
Yeah, I was trying to keep some contract with this function and the method inputs being 1:1 with Kliento calls. But doesn't have to be so.